hugolib/page__per_output.go

```diff
@@ -128,6 +128,17 @@ func newPageContentOutput(p *pageState) func(f output.Format) (*pageContentOutpu
 							cp.summary = helpers.BytesToHTML(summary)
 						}
 					}
+				} else if cp.p.m.summary != "" {
+					html := cp.p.s.ContentSpec.RenderBytes(&helpers.RenderingContext{
+						Content: []byte(cp.p.m.summary), RenderTOC: false, PageFmt: cp.p.m.markup,
+						Cfg:        p.Language(),
+						DocumentID: p.File().UniqueID(), DocumentName: p.File().Path(),
+						Config: cp.p.getRenderingConfig()})
+					// Strip enclosing <p>
+					html = []byte(strings.TrimSpace(string(html)))
```

Comment:
These multiple `[]byte` conversions look slightly ineffective. I would probably just convert it to a `string` as a first step (as that is what it's going to end up as) and do the trimming on that ...

That said, we've had some discussions on this before, whether we should trim `p` tags or not.

I'm not totally sure what we do in the "summary split" case, but the code I read here will not work as a general case, so I think that needs to be adjusted/removed. The main problem is the handling of multiple paragraphs. I suggest that we get it in line with the other Markdown summary option.

Reply:
Second thought, if you pull the logic in 

https://github.com/gohugoio/hugo/blob/950295516da882dcc51d83f70835dde230a0b4d6/tpl/transform/transform.go#L116

Into some common function, and then use that.

Reply:
Would you prefer the common function to remain in `transform.go` or move to `helpers/`?

Reply:
Move it to helpers/content.go (it cannot live in transform.go anyway because that will create a cyclic reference)

Reply:
Add it as a method on the ContentSpec struct to make less global helper funcs.

Reply:
Done.  I was tempted to let the function trim all HTML passed in to it but that would change existing output so I stuck to principle of least surprise and only do so when the input is recognised as a single paragraph.